From b397fd4b95a5d097e71318a5a451700c6d3cff7a Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Thu, 14 Sep 2023 12:09:33 +0200
Subject: [PATCH 8/8] Add LuneOS modules and machine specific cmake file to
 build

Include all the LuneOS modules and the machine specific cmake file.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 CMakeLists.txt     |  7 ++++++-
 src/CMakeLists.txt | 38 +++++++++++++++++++++++++++++++++++++-
 2 files changed, 43 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 411ff94..0e3d688 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,5 +32,10 @@ webos_add_compiler_flags(ALL ${PMLOG_CFLAGS_OTHER})
 
 include_directories(include/internal)
 
-webos_nyx_module_provider(OW BATTERY CHARGER DEVICEINFO OSINFO SYSTEM DISPLAY SECURITY SECURITY2 GPS CEC)
+webos_nyx_module_provider(OW BATTERY CHARGER DEVICEINFO OSINFO SYSTEM DISPLAY SECURITY SECURITY2 GPS CEC MSMMTP ALS LED HAPTICS KEYS TOUCHPANEL TOUCHPANEL_MTDEV)
+
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/src/machine.cmake)
+	include(${CMAKE_CURRENT_SOURCE_DIR}/src/machine.cmake)
+endif()
+
 add_subdirectory(src)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d435c35..1aed92a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -36,7 +36,7 @@ pkg_check_modules(GIO REQUIRED gio-2.0)
 include_directories(${GIO_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${GIO_CFLAGS_OTHER})
 
-if(NYXMOD_OW_BATTERY OR NYXMOD_OW_CHARGER)
+if(NYXMOD_OW_BATTERY OR NYXMOD_OW_CHARGER OR NYXMOD_OW_MSMMTP OR NYXMOD_OW_LED OR NYXMOD_OW_HAPTICS)
     pkg_check_modules(UDEV REQUIRED libudev)
     include_directories(${UDEV_INCLUDE_DIRS})
     webos_add_compiler_flags(ALL ${UDEV_CFLAGS_OTHER})
@@ -90,3 +90,39 @@ endif()
 if(NYXMOD_OW_CEC)
     add_subdirectory(cec)
 endif()
+
+# Modules added for LuneOS
+if(NYXMOD_OW_TOUCHPANEL_MTDEV)
+    pkg_check_modules(MTDEV REQUIRED mtdev)
+    include_directories(${MTDEV_INCLUDE_DIRS})
+    webos_add_compiler_flags(ALL ${MTDEV_CFLAGS_OTHER})
+    add_subdirectory(touchpanel_mtdev)
+endif()
+
+if(NYXMOD_OW_TOUCHPANEL)
+    add_subdirectory(touchpanel)
+endif()
+
+if(NYXMOD_OW_MSMMTP)
+    add_subdirectory(msm_mtp)
+endif()
+
+if(NYXMOD_OW_ALS)
+    add_subdirectory(als)
+endif()
+
+if(NYXMOD_OW_KEYS)
+    add_subdirectory(keys)
+endif()
+
+if(NYXMOD_OW_LED)
+    add_subdirectory(led_controller)
+endif()
+
+if(NYXMOD_OW_HAPTICS)
+    add_subdirectory(haptics_timedoutput)
+endif()
+
+if(NYXMOD_OW_GPS)
+    add_subdirectory(gps)
+endif()
-- 
2.41.0.windows.3

